dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/SerializableSchedule.java

```diff
@@ -215,44 +296,86 @@ public boolean equals(Object o) {
 
         //Compare fields taking into account possible null values
         boolean startupBool = Boolean.compare(this.startup, c.startup) == 0;
-        boolean validDays = this.validDays == null ? c.validDays == null : this.validDays.equals(c.validDays);
+        boolean validDaysBool = this.validDays == null ? c.validDays == null : (this.validDays.containsAll(c.validDays) && c.validDays.containsAll(this.validDays));
         boolean startTimeBool = this.startTime == null ? c.startTime == null : this.startTime.equals(c.startTime);
         boolean endTimeBool = this.endTime == null ? c.endTime == null : this.endTime.equals(c.endTime);
+        boolean timezoneBool = this.timezone == null ? c.timezone == null : this.timezone.equals(c.timezone);
 
-        return startupBool && validDays && startTimeBool && endTimeBool;
+        return startupBool && validDaysBool && startTimeBool && endTimeBool && timezoneBool;
     }
 
     /**
-     * <pre>
-     * Method to determine if we are within the scheduled day and time within a margin of time.
-     * The margin provided should represent the number of seconds between this call and the next time you plan to check the schedule.
+     * Method to determine if the current server time is within the schedule represented by this object.
+     * Note: This method will automatically adjust to use the schedule's timezone, and not the server's timezone.
+     *
+     * @return boolean - true the time provided is within the scheduled represented by this object, false otherwise.
+     */
+    public boolean withinSchedule() {
+        return withinSchedule(ZonedDateTime.now());
+    }
+
+    /**
+     * Method to determine if a LocalDateTime is within the schedule represented by this object.
+     * Note: This method assumes that the LocalDateTime provided represents a ZonedDateTime within the timezone of this schedule.
      *
-     * Example1: if the schedule is "MON 8:00" and you are checking the schedule every 6 minutes then calling
-     * this method between 7:57 and 8:03 will return true on Monday.
+     * @param time - LocalDateTime
+     * @return boolean - true the time provided is within the scheduled represented by this object, false otherwise.
+     */
+    public boolean withinSchedule(LocalDateTime time) {
+        return withinSchedule(time.atZone(timezone));
```

Comment:
I wouldn't expect the consumer of the schedule to want to supply a time that they have converted to be in the configured zone.  It seems more likely they will have something like Instant.now() or ZonedDateTime.now(), which you already have a method for.

This is also error prone.  I could see someone supplying LocalDateTime.now() which is computed in the server's time zone then being interpreted by this method as though the value were the configured time zone.